index_api-v2.md

```diff
@@ -0,0 +1,191 @@
+
+The record of a node in the index's database can be in one of five possible states: `received`, `validated`, `validation_failed`, `posted`, `post_failed` or `deleted`. The node will only be discoverable in the index when it has the status of `posted` or `deleted`.
+
+This endpoint enables a Murmurations profile generator to get and present an update to the node operator as to the status of the node profile after it has been submitted to the index (i.e., when using `POST /nodes`).
+
+#### Input
+
+- the `node_id` of the node's profile that is currently in the index
+
+#### Output
+
+Detailed response examples can be seen in the [Open API specification](https://app.swaggerhub.com/apis-docs/MurmurationsNetwork/IndexAPI/2.0.0#/Node%20Endpoints/get_nodes__node_id_) under the _Responses_ section for `200 OK` response code.
+
+##### Success
+
+- Confirmation of status (e.g., `received`, `validated`, `posted` or `deleted`)
+
+##### Error
+
+###### Validation Failed
+
+- Could not download profile from `profile_url`
+- Could not validate profile against one or more schemas
+- Could not find one or more schemas in library for validation
+
+###### Post Failed
+
+- Could not post validated profile to the index (internal service issue)
+
+### [`DELETE /nodes/{node_id}`](https://app.swaggerhub.com/apis-docs/MurmurationsNetwork/IndexAPI/2.0.0#/Node%20Endpoints/delete_nodes__node_id_)
+
+Node operators will use the `DELETE /nodes/{node_id}` endpoint to remove their profile from the index when they no longer want it listed.
+
+#### Input
+
+- the `node_id` of the profile that is currently in the index
+  - The profile must no longer be available at the `profile_url` (i.e., should return a `404 Not Found` error when accessing the URL) or it will not be removed from the index
+
+#### Output
+
+##### Success
+
+- Confirmation of removal (e.g., `200 OK` success status response code)
+
+##### Error
+
+- Return a `404 Not Found` error when the `node_id` does not exist in the index
+- Return a `400 Bad Request` error when the profile is still available at its `profile_url` on the web
+
+## Aggregator Endpoints
+
+### [`GET /nodes`](https://app.swaggerhub.com/apis-docs/MurmurationsNetwork/IndexAPI/2.0.0#/Aggregator%20Endpoints/get_nodes)
+
+<!-- TODO - add this back if/when it happens
+> :lock: The `GET /nodes` endpoint will eventually require an API key in order to prevent unauthorized harvesting of data from the index.
+ -->
+
+Aggregators can search for nodes based on the schemas nodes use, when the nodes were last validated by the index (i.e., for recent changes to node profiles), and by geolocating nodes within a certain range (e.g., _10km_ or _6mi_) from a specific location or finding them based on the town/city, country, etc. Searching for deleted nodes is also possible so that aggregators can keep their records updated as nodes remove themselves from the index. Finally, nodes can associate tags to their profiles to enable searching by specific keywords.
```

Comment:
Suggestion (last_validated > updated, and adding primary_url)

Aggregators can search for nodes based on the schemas nodes use, when the nodes were last updated, and by geolocating nodes within a certain range (e.g., _10km_ or _6mi_) from a specific location or finding them based on the town/city, country, etc. Searching for deleted nodes is also possible so that aggregators can keep their records updated as nodes remove themselves from the index. Finally, nodes can associate tags to their profiles to enable searching by specific keywords, and nodes can be queried based on their `primary_url` value.

Reply:
Done.